File: Queue/Problem1 - 933.java
package Queue;

import java.util.LinkedList;
import java.util.Queue;

class RecentCounter {
    private Queue<Integer> q;

    public RecentCounter() {
        q = new LinkedList<>();
    }

    public int ping(int t) {
        q.offer(t);
        while (!q.isEmpty() && q.peek() < (t - 3000)) {
            q.poll();
        }

        return q.size();
    }
}

File: Queue/Problem1 - 933.cpp
#include <bits/stdc++.h>
using namespace std;

/* optimal approach - using queue */
class RecentCounter {
private: 
    queue<int> q; 
public:
    RecentCounter() { }
    
    int ping(int t) {
        q.push(t);   
        while(!q.empty() && q.front() < (t - 3000)) {
            q.pop(); 
        }

        return q.size(); 
    }
};